### DISCUSSION TITLE
Getting events from public client

### DISCUSSION THREAD

**Discussion by Anup9444** [wevm/viem#937] [Question] [answered]
Hi,
i am trying to figure out what are the things i am missing here,i am trying to get event logs but i am receiving empty array
i guess i am doing all correct but could not able to get any events.. i tried to using `createContractEventFilter` , 
i am not  geting any logs

import {  usePublicClient } from "wagmi";
 const publicClient = usePublicClient({ chainId })
![image](https://github.com/wagmi-dev/viem/assets/110972959/bd7dcc23-6cf3-46ed-8ed0-833230999dd4)
this is the ABI 
![image](https://github.com/wagmi-dev/viem/assets/110972959/7dcc6bed-a80d-4e16-adae-2466fdf46b97)

**Comment by Anup9444** [comment, marked as answer]
i made some smalll changes and it worked now 
![image](https://github.com/wagmi-dev/viem/assets/110972959/8f287dd1-354b-4f20-8ec1-da9067ad8dbe)

**Comment by ECBSJ** [comment]
I've also been having issues with `publicClient.createContractEventFilter` and `contractInstance.createEventFilter`. None of them are working for me.

**Comment by zigzag-way** [comment]
Im calling getLogs but in return I get empty array without any errors, with misc contracts I tried it.

**Reply by jxom** [reply]
Can you share a minimal reproducible example? Hard to tell what's going on without any info.

**Reply by narekpetrosyan** [reply]
```
 const contract = getContract({
        abi: Factory__Abi,
        address: FACTORY_ADDRESS,
        client: { 
          public: publicClient, 
        } 
      })

      const filter = await contract.createEventFilter.PairCreated()
      const logs = await publicClient.getFilterLogs({filter});

      console.log(logs);
```
